viewing_party/party.py

```diff
@@ -0,0 +1,201 @@
+def create_movie(movie_title, genre, rating):
+    '''
+    function builds a dictionary with three key value 
+    pairs where each value is the relevant parameter for each movie
+    '''
+    if movie_title == None or genre == None or rating == None:
+        return None
+    new_movie = {
+        "title": movie_title,
+        "genre": genre,
+        "rating": rating
+        }
+    print(new_movie)
+    return new_movie
+
+
+def add_to_watched(user_data, movie):
+    '''
+    function adds a specific movie title to the users 
+    list of watched movies
+    '''
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    '''
+    function adds a specific movie title to the users 
+    list of movies to wacth
+    '''
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    '''
+    function checks if movie is in users watchlist. if so, adds 
+    movie to users watched list and removes it from users watchlist. 
+    returns updated user lists
+    '''
+    movies_to_remove = []
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            movies_to_remove.append(movie)
+
+    if len(movies_to_remove) == 0:
+        return user_data
+    else:
+        for movie in movies_to_remove:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            
+    return user_data
+
+
+def get_watched_avg_rating(user_data):
+    '''
+    function gets average rating for the movies that are in 
+    users list fo watched movies
+    '''
+    ratings = []
+    if len(user_data["watched"]) == 0:
+        return 0.0
+    else:
+        for movie in user_data["watched"]:
+            ratings.append(movie["rating"])
+        num_of_movies = len(ratings)
+        ratings_total = sum(ratings)
```

Comment:
nice! one thing we can look at here is we've got a for loop grabbing each rating, but then we have another for loop under the hood summing up the ratings.

let's add them together:

```suggestion
        ratings_total = 0.0
        for movie in user_data["watched"]:
            ratings_total += movie["rating"]
        num_of_movies = len(ratings)
```